### ISSUE TITLE
Missing tags for corner nodes in rectangular mesh generator

### ISSUE THREAD

**Issue by Galaxxy31** [bcube-project/Bcube.jl#56]
In the function `_rectangle_quad_mesh` (src/mesh/mesh_generator.jl), tags are assigned as follows :

```julia
if ix == 1 # West
    push!(tag2nodes[4], iglob)
elseif ix == nx # East
    push!(tag2nodes[3], iglob)
elseif iy == 1
    push!(tag2nodes[2], iglob)
elseif iy == ny
    push!(tag2nodes[1], iglob)
end
```

Nodes on the corners of the mesh should have two tags since they belong to both boundaries, which is not the case with the current implementation.
A possible solution would be :

```julia
if ix == 1 # West
    push!(tag2nodes[4], iglob)
end
if ix == nx # East
    push!(tag2nodes[3], iglob)
end
if iy == 1 # South
    push!(tag2nodes[2], iglob)
end
if iy == ny # North
    push!(tag2nodes[1], iglob)
end
```

It remains to find out how to deal with a superposition of Dirichlet boundaries conditions on these nodes.

**Comment by bmxam** [comment]
I've proposed a PR with your fix.

Up to now, Dirichlet conditions are only implemented on "faces" (elements of codimension 1) and the API doesn't have any high-level-function to apply Dirichlet conditions on "nodes" (elements of codimension 2). But it is possible to develop such a function with what we have in `Bcube`.

@Galaxxy31 if you already have something like this, don't hesitate to propose a PR. On the other hand if need it but don't know how to do it, we can take a look at it together.

**Comment by lokmanb** [comment]
@Galaxxy31 could you tell us more about your intended use of these multiple tags please?

**Comment by Galaxxy31** [comment]
I don't really have a use for these multiple tags.
The point I was trying to make is that with the current implementation, some boundaries conditions might not be completely defined.

For example, I want to model heat diffusion in a rectangle at 300K, whose southern edge has an imposed temperature of 400K. Since the nodes in the corners don't have the "south" tag, the boundary condition I want to impose is not respected (see figure below).

![issue56](https://github.com/bcube-project/Bcube.jl/assets/64801148/36187b1b-cf6c-4db4-9e78-7d4e5341d930)

Here, all the nodes on the south edge have a value of 400K, but those on the corners have a value of 300K (as if they didn't belong to the south edge). The problem would be the same with the north edge but wouldn't be the case if the boundary condition were imposed on the west (or east) edge.

Dealing with two potential boundaries conditions on the corners is another question...

**Comment by bmxam** [comment]
the initial problem described has been solved in #59
